### PR TITLE
Add asyncapi playground to dev portal.

### PR DESCRIPTION
**Pull request by sachithKay**
- Adds a component to view the definition of an Async API
- Uses official react component for Async APIs.
- Does not implement new backend endpoints.
- Refactors the existing getAPIbyId to add the asyncapi.yml as the API definition.
- asyncapi.yml is read from the registry if the API type is (WEBSUB || SSE || ES)

UI Change
Screen Shot 2021-01-29 at 9 23 09 PM

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java [hunk 1]
```diff
@@ -1225,6 +1226,35 @@ public String getOpenAPIDefinition(String apiId, String tenantDomain) throws API
         return definition;
     }
 
+    @Override
+    public String getAsyncAPIDefinition(Identifier apiId) throws APIManagementException {
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java:1230]: This method is not needed. lets use the getAsyncAPIDefinition(String apiId, String tenantDomain) method only

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIProvider.java [hunk 2]
```diff
@@ -256,6 +256,12 @@ public String getOpenAPIDefinition(Identifier apiId) throws APIManagementExcepti
         return super.getOpenAPIDefinition(apiId);
     }
 
+    @Override
+    public String getAsyncAPIDefinition(Identifier apiId) throws APIManagementException {
+        checkAccessControlPermission(apiId);
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIProvider.java:261]: This is not needed. accesscontrol is checked with the getAPI() call done before this

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java (hunk 1 above)
Comment (sachithKay) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java]: Removed this method from APIManager interface. Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIProvider.java (hunk 2 above)
Comment (sachithKay) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIProvider.java]: Removed override. All APIConsumers and providers will inherit the original method from the super class.

**Review by chamilaadhi — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import '@asyncapi/react-component/lib/styles/fiori.css';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import Typography from '@material-ui/core/Typography';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import { ApiContext } from '../../ApiContext';
+
+const styles = (theme) => ({
+    paper: {
```
Comment (chanaka3d) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx:30]: We can remove all these style definitions.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    static contextType = ApiContext;
+
+    constructor(props) {
+        super(props);
+        this.state = { spec: '' };
+    }
+
+    componentDidMount() {
+        const context = this.context;
+        this.setState({ spec: context.api.apiDefinition });
```
Comment (chanaka3d) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx:149]: This state is not required.
You can directly use apiDefinition from context.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            <>
+                <Typography variant='h4' className={classes.titleSub}>
+                    <FormattedMessage
+                        id='Apis.Details.Async.Definition.title'
+                        defaultMessage='AsyncAPI Specification'
+                    />
+                </Typography>
+                <Grid container spacing={1} className={classes.editorRoot}>
+                    <Grid item className={classes.editorPane}>
+                        <AsyncApiComponent schema={spec} />
```
Comment (chanaka3d) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx:165]: No need to get the spec from the context and pass. You can simply get the same within the AsyncAPIComponent.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        minHeight: '100vh',
+        overflow: 'auto',
+    },
+    editorRoot: {
+        height: '100%',
+    },
+});
+
+
+class AsyncApiDefinitionUI extends Component {
```
Comment (chanaka3d) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx:139]: You can convert this to a functional component.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/index.jsx [hunk 7]
```diff
@@ -361,6 +363,15 @@ class Details extends React.Component {
         this.setState({ api });
     }
 
+    isAsyncAPI(api) {
+        if (api) {
+            if (api.type === 'WS' || api.type === 'WEBSUB' || api.type === 'SSE') {
```
Comment (chanaka3d) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/index.jsx:368]: You can simply return (api.type === 'WS' || api.type === 'WEBSUB' || api.type === 'SSE')

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx (hunk 3 above)
Comment (sachithKay) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx]: removed

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx (hunk 4 above)
Comment (sachithKay) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/index.jsx (hunk 7 above)
Comment (sachithKay) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/index.jsx]: done

**Review by chanaka3d — APPROVED** [review]

**Pull request merged**